src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -244,6 +244,11 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
             if prop in forbidden:
                 forbidden.remove(prop)
 
+    # set default storage SKU if not provided and using an image based OS
+    if not namespace.storage_sku and namespace.storage_profile \
+        not in [StorageProfile.ManagedSpecializedOSDisk, StorageProfile.SASpecializedOSDisk]:
+        namespace.storage_sku = 'Standard_LRS'
```

Comment:
The default one should be `Premium_LRS` for VM and `Standard_LRS` for VMSS. Yep, it was per discussion with CRP team 2 weeks ago